scripts/datacube.py

```diff
@@ -423,28 +422,54 @@ def convert_attrs_and_coords_objects_to_str(data):
 @click.option(
     "--subset",
     required=False,
+    default="",
+    help="For debugging, subset x and y to this pixel window as a comma"
+    "separated string of 4 integers.",
```

Comment:
I see that this is for debugging only, but just to clarify, this option is for cropping the tiles/chips to a specified height and width?

Reply:
I see this is creating confusion, I'll try to work this better in the help string.

The goal of this is to reduce the amount of data when processing in test mode. Normally, the script would download the entire scene, which is 10k pixels squared. Whit this one can reduce the amount of pixels downloaded and passed to the tiler script. So this is cutting a window from the scene data. The tiler will then work on this smaller input, but be unaltered otherwise.

Reply:
Not to make it more complex, but I understand we download the entire MGRS tile, even when many of the  chips might be outside of the requested AOI. If we drop chips outside of the AoI, we can just use a point or small polygon for testing.

Reply:
That is correct and would be the best approach, but for that the datacube.py script needs much more refactoring. Currently it's fully focused on downloading the entire MGRS tile. Filtering that spatially will require passing the AOI itself and handling spatial intersections, so that indeed is quite a bit more complicated.

For now I can state also that more clearly, that the AOI approach right now means getting data for full MGRS tiles that intersect with the AOI.